src/MaxLikelihoodFit.cc

```diff
@@ -525,7 +569,7 @@ void MaxLikelihoodFit::getNormalizations(RooAbsPdf *pdf, const RooArgSet &obs, R
         //out.addOwned(*(new RooConstVar(pair->first.c_str(), "", pair->second.norm->getVal())));
         if (fOut != 0 && saveShapes_ && pair->second.obs.getSize() == 1) {
             RooRealVar *x = (RooRealVar*)pair->second.obs.at(0);
-            TH1* hist = pair->second.pdf->createHistogram("", *x);
+            TH1* hist = pair->second.pdf->createHistogram("", *x, pair->second.isfunc ? RooFit::Extended(false) : RooCmdArg::none());
```

Comment:
For some reason createHistogram complains a lot if you call on it a non-pdf object without the `RooFit::Extended(false)` option.